### PR TITLE
API Property-Based tests

### PR DESCRIPTION
**Pull request by clearydude**
This pull request adds a suite of property-based tests for the API. The tests open a vcan socket so that they can intercept and send messages and ensure that the API responds correctly. They also check that spoof values are within range and correctly calculated for any input. Brake tests are included for both supported vehicles.

### CODE REVIEW

File: api/tests/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,17 @@
+cmake_minimum_required(VERSION 2.8)
+
+project(api-tests)
+
```
Comment (austinjmorlan) [inline api/tests/CMakeLists.txt]: Is it possible for CMake to run the `initialize_vcan` script instead of requiring the user to do it? It's easy to forget and then you see failed tests and have momentary panic.

File: README.md [hunk 2]
```diff
@@ -410,6 +412,22 @@ running 0 tests
...
+```
+cd api/tests
+mkdir build
+cd build
+cmake .. -DKIA_SOUL=ON
+```
+
+The tests can then be run using `make run-api-property-tests`. For the tests to run, you must have a vcan interface initialized. The test folder contains a script `initialize_vcan.sh` which can be used to set it up automatically.
+
+These tests run against the 1.17.0 Rust toolchain, so it will need to be installed via **rustup** if it isn't already.
```
Comment (austinjmorlan) [inline README.md]: Can these two lines be moved up above to be below the `Rust installation guide` section? I think it's easy to miss this.

File: api/tests/property/include/wrapper.h [hunk 3]
```diff
@@ -0,0 +1,8 @@
+#include "can_protocols/brake_can_protocol.h"
```
Comment (austinjmorlan) [inline api/tests/property/include/wrapper.h]: `oscc.h` now includes the CAN protocol headers and `vehicles.h`, so you don't have to explicitly include these.

File: api/tests/property/tests/brake_tests.rs [hunk 4]
```diff
@@ -0,0 +1,239 @@
...
+            Err(why) => TestResult::discard(),
+            Ok(frame) => {
+                let brake_command_msg = get_brake_command_msg_from_buf( frame.data() );
+
+                let spoof_high = brake_command_msg.spoof_value_high as u32;
+                let spoof_low = brake_command_msg.spoof_value_low as u32;
+
+                TestResult::from_bool( 
+                    (spoof_high <= BRAKE_SPOOF_HIGH_SIGNAL_RANGE_MAX) && 
+                    (spoof_high >= BRAKE_SPOOF_HIGH_SIGNAL_RANGE_MIN) && (spoof_low <= BRAKE_SPOOF_LOW_SIGNAL_RANGE_MAX) && (spoof_low >= BRAKE_SPOOF_LOW_SIGNAL_RANGE_MIN))
```
Comment (austinjmorlan) [inline api/tests/property/tests/brake_tests.rs]: Missing a new line here after the `&&`.

File: api/tests/property/tests/steering_tests.rs [hunk 5]
```diff
@@ -0,0 +1,189 @@
...
+        Err(why) => TestResult::discard(),
+        Ok(frame) => {
+            let steering_command_msg = get_steering_command_msg_from_buf( frame.data() );
+
+            let spoof_high = steering_command_msg.spoof_value_high as u32;
+            let spoof_low = steering_command_msg.spoof_value_low as u32;
+
+            TestResult::from_bool( 
+                (spoof_high <= STEERING_SPOOF_HIGH_SIGNAL_RANGE_MAX) && 
+                (spoof_high >= STEERING_SPOOF_HIGH_SIGNAL_RANGE_MIN) && (spoof_low <= STEERING_SPOOF_LOW_SIGNAL_RANGE_MAX) && (spoof_low >= STEERING_SPOOF_LOW_SIGNAL_RANGE_MIN))
```
Comment (austinjmorlan) [inline api/tests/property/tests/steering_tests.rs]: Missing newline here after `&&`.

File: api/tests/CMakeLists.txt (hunk 1 above)
Comment (k-cleary) [reply api/tests/CMakeLists.txt]: 881472f

File: README.md (hunk 2 above)
Comment (k-cleary) [reply README.md]: 881472f

File: api/tests/property/include/wrapper.h (hunk 3 above)
Comment (k-cleary) [reply api/tests/property/include/wrapper.h]: 881472f

File: api/tests/property/tests/brake_tests.rs (hunk 4 above)
Comment (k-cleary) [reply api/tests/property/tests/brake_tests.rs]: 881472f

File: api/tests/property/tests/steering_tests.rs (hunk 5 above)
Comment (k-cleary) [reply api/tests/property/tests/steering_tests.rs]: 8e62735

File: api/include/vehicles/kia_soul_ev.h [hunk 6]
```diff
@@ -147,7 +147,7 @@ typedef struct
  *
  * Equal to \ref BRAKE_SPOOF_LOW_SIGNAL_VOLTAGE_MIN * \ref STEPS_PER_VOLT.
  */
-#define BRAKE_SPOOF_LOW_SIGNAL_RANGE_MIN ( 273 )
```
Comment (adrienthebo) [inline api/include/vehicles/kia_soul_ev.h]: Are these changes desired? They seem unrelated to API property tests.

File: api/tests/property/src/lib.rs [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+pub fn skip_enable_frames(socket: &CANSocket) {
+    socket.read_frame();
+    socket.read_frame();
+    socket.read_frame();
+}
+
+pub fn init_socket() -> CANSocket {
+    // initialize our PBT socket on the vcan0
+    let socket_result = CANSocket::open("vcan0");
+    match socket_result {
```
Comment (adrienthebo) [inline api/tests/property/src/lib.rs]: Would [`Result.expect`](https://doc.rust-lang.org/std/result/enum.Result.html#method.expect) be appropriate here?
